### PR TITLE
Frontend Enhancements

### PR DESCRIPTION
**Pull request by subhamX**
## Description
The following PR revamps the frontend design and enhances the mobile and smaller devices experience with the linked commons. It also modularizes and updates the code documentation.

## Screenshots
The demo version can be accessed [here](http://ccdataviz.netlify.app/).

## Checklist
 
- [x] My pull request has a descriptive title (not a vague title like `Update
  index.md`).
- [x] My pull request targets the *default* branch of the repository (`main` or `master`).
- [x] My commit messages follow [best practices][best_practices].
- [x] My code follows the established code style of the repository.
- [ ] I added tests for the changes I made (if applicable).
- [x] I added or updated documentation (if applicable).
- [x] I tried running the project locally and verified that there are no
  visible errors.

[best_practices]:https://gist.github.com/robertpainsi/b632364184e70900af4ab688decf6f53

## Developer Certificate of Origin

<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>

### CODE REVIEW

File: frontend/.gitignore [hunk 1]
```diff
@@ -4,7 +4,7 @@
 /node_modules
 /.pnp
 .pnp.js
-
+.env
```
Comment (mathemancer) [inline frontend/.gitignore]: Why was this added?  For auto deployment, we need the environment to be picked up when the image is build _with no modification_ (since this will happen on github)

**Review by mathemancer — CHANGES_REQUESTED** [review]
Overall, this looks great, thanks!

However, I'm concerned about ignoring the `.env` file for the frontend.  For the back end, that isn't an issue, but for the front we need the API host variable at build time, no?

File: frontend/.gitignore (hunk 1 above)
Comment (subhamX) [reply frontend/.gitignore:7]: I might have added it by mistake. Removing it right away.

**Comment by subhamX** [comment]
> Overall, this looks great, thanks!
> 
> However, I'm concerned about ignoring the `.env` file for the frontend. For the back end, that isn't an issue, but for the front we need the API host variable at build time, no?

@mathemancer, even for the frontend, we can define env variables in the machine and the `react-scripts` can use them. We just need to make sure that the variable name starts from `REACT_APP_`, and there shouldn't be any `.env` which is overriding the value.

I added `.env` in gitignore as I was trying to merge the `.env` files of backend and frontend and place it in the project root directory. [Ref:- [Link](https://github.com/creativecommons/cccatalog-dataviz/pull/39#discussion_r474921581)] I dropped the idea as I didn't find any reasonable way to do the merge, but I forgot to remove the line from the `.gitignore`. I will be more cautious in the future!

**Review by mathemancer — APPROVED** [review]
Looks good to me.

Let's give @soccerdroid a bit of time to look it over before merging.

**Review by soccerdroid — APPROVED** [review]
